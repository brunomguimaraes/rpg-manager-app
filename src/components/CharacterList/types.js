// @flow

import type { Character } from "~/components/Character";
export type Props = {
  characters: Array<Character>
};
