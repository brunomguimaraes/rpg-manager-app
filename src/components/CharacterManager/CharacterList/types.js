// @flow

import type { Character } from "~/components/CharacterManager";
export type Props = {
  gameId: number,
  scrollable: boolean,
  characters: Array<Character>
};
