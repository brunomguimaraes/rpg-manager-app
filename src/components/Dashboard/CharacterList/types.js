// @flow

import type { Character } from "~/components/Dashboard/Character";
export type Props = {
  characters: Array<Character>
};
