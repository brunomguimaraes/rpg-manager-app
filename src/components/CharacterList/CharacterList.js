// @flow

import React from "react";
import { Scrollbars } from "react-custom-scrollbars";
import Character from "~/components/Character";
import type { Props } from "./types";
import styles from "./CharacterList.module.scss";

export default function CharacterList({ characters }: Props) {
  return (
    <Scrollbars autoHide>
      <ul className={styles.characterList}>
        {characters.map(character => (
          <li key={character.id}>
            <Character {...character} />
          </li>
        ))}
      </ul>
    </Scrollbars>
  );
}
