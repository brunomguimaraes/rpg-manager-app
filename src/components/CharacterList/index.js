// @flow

export { default } from "./CharacterList";
