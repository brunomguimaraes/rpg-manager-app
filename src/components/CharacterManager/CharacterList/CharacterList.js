// @flow

import React from "react";
import Scrollable from "~/components/Scrollable";
import Character from "../Character";
import type { Props } from "./types";
import styles from "./CharacterList.module.scss";

export default function CharacterList({
  characters,
  scrollable,
  gameId
}: Props) {
  return (
    <Scrollable shouldScroll={scrollable}>
      <ul className={styles.characterList}>
        {characters.map(character => (
          <li key={character.id}>
            <Character gameId={gameId} {...character} />
          </li>
        ))}
      </ul>
    </Scrollable>
  );
}
