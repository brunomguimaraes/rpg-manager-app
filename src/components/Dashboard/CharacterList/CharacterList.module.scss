@import "~styles/variables";

.characterList {
  width: 100%;
  margin: 0;
  padding: 0;
  display: block;
  overflow: hidden;
  padding: $spacer;
  list-style-type: none;
  box-sizing: border-box;
  & > li {
    padding: $spacer;
  }
}
