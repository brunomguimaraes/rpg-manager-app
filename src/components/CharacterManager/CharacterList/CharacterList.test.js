import React from "react";
import CharacterList from "./CharacterList";

const props = {
  characters: [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }]
};

test("CharacterList", () =>
  expect(shallow(<CharacterList {...props} />)).toMatchSnapshot());
