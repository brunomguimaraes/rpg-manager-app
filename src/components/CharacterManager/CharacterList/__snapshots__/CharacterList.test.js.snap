// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList 1`] = `
<Scrollable>
  <ul
    className="characterList"
  >
    <li
      key="1"
    >
      <Character
        id={1}
      />
    </li>
    <li
      key="2"
    >
      <Character
        id={2}
      />
    </li>
    <li
      key="3"
    >
      <Character
        id={3}
      />
    </li>
    <li
      key="4"
    >
      <Character
        id={4}
      />
    </li>
  </ul>
</Scrollable>
`;
